src/fiber/gossip.rs

```diff
@@ -2442,6 +2552,28 @@ where
                 }
             }
 
+            GossipActorMessage::PruneStaleGossipMessages(timestamp) => {
+                if state.has_active_prune_task || timestamp == 0 {
+                    return Ok(());
+                }
+                state.has_active_prune_task = true;
+                let task_tracker = new_tokio_task_tracker();
+                let cancellation_token = new_tokio_cancellation_token();
+                let store = state.store.get_store().clone();
+                let myself = myself.clone();
+                // Spawning a subtask to avoid blocking normal message processing of the actor.
```

Comment:
I don't think we need to spawn a new task here, the prune processing is lightweight, it should be completed in several ms.